core/src/main/scala/kafka/log/Log.scala

```diff
@@ -609,24 +704,22 @@ class Log(val dir: File,
    * logSegment will be rolled if one of the following conditions met
    * <ol>
    * <li> The logSegment is full
-   * <li> The maxTime has elapsed
+   * <li> The maxTime has elapsed since the largest timestamp in the segment (or since the create time if there is no
+   * message having a timestamp)
```

Comment:
The comment is not accurate. The rolling is now based on the timestamp of the first message.